app/src/main/java/org/p2p/wallet/striga/sms/onramp/StrigaOnRampSmsInputFragment.kt

```diff
@@ -0,0 +1,70 @@
+package org.p2p.wallet.striga.sms.onramp
+
+import androidx.fragment.app.setFragmentResult
+import android.os.Bundle
+import android.view.View
+import org.koin.android.ext.android.inject
+import org.koin.core.parameter.parametersOf
+import org.koin.core.qualifier.named
+import org.p2p.wallet.R
+import org.p2p.wallet.auth.model.GatewayHandledState
+import org.p2p.wallet.auth.model.PhoneNumber
+import org.p2p.wallet.auth.model.RestoreFailureState
+import org.p2p.wallet.smsinput.BaseSmsInputFragment
+import org.p2p.wallet.smsinput.SmsInputContract
+import org.p2p.wallet.striga.di.StrigaWalletModule
+import org.p2p.wallet.striga.sms.error.StrigaSmsErrorFragment
+import org.p2p.wallet.striga.sms.error.StrigaSmsErrorViewType
+import org.p2p.wallet.striga.wallet.models.ids.StrigaWithdrawalChallengeId
+import org.p2p.wallet.utils.args
+import org.p2p.wallet.utils.emptyString
+import org.p2p.wallet.utils.popBackStack
+import org.p2p.wallet.utils.replaceFragment
+
+class StrigaOnRampSmsInputFragment : BaseSmsInputFragment() {
+
+    companion object {
+        const val ARG_TITLE_AMOUNT = "ARG_TITLE_AMOUNT"
+        const val ARG_CHALLENGE_ID = "ARG_CHALLENGE_ID"
+        val REQUEST_KEY: String = StrigaOnRampSmsInputFragment::class.java.name
+    }
+
+    private val titleAmount: String by args(ARG_TITLE_AMOUNT)
+    private val challengeId: StrigaWithdrawalChallengeId by args(ARG_CHALLENGE_ID)
```

Comment:
wow
that's new
what is a challenge?

Reply:
this model was introduced with initiate/withdrawal api call, this id is just a sort of internal transaction-id in fact

Reply:
so maybe we can call it just StrigaWithdrawalTransactionId or StrigaWithdrawalId?

Reply:
i'm not sure that is a good idea to change it to `transaction id` because the `challenge` actually means that the system is asking for a user authentication and the `challengeId` represents some temporary key to use in this authentication